vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -158,32 +216,11 @@ def pre_build(self):
                 )
             ]
 
-    def __call__(self, target_to_data_frame: dict[ModelID, pd.DataFrame]):
-        # Only relevant for a dynamic filter.
-        # Although targets are fixed at build time, the validation logic is repeated during runtime, so if a column
-        # is missing then it will raise an error. We could change this if we wanted.
-        # Call this from actions_utils
-        targeted_data = self._validate_targeted_data(
-            {target: data_frame for target, data_frame in target_to_data_frame.items() if target in self.targets},
-            eagerly_raise_column_not_found_error=True,
-        )
-
-        if (column_type := self._validate_column_type(targeted_data)) != self._column_type:
-            raise ValueError(
-                f"{self.column} has changed type from {self._column_type} to {column_type}. A filtered column cannot "
-                "change type while the dashboard is running."
-            )
-
-        # TODO: when implement dynamic, will need to do something with this e.g. pass to selector.__call__.
-        # if isinstance(self.selector, SELECTORS["numerical"] + SELECTORS["temporal"]):
-        #     options = self._get_options(targeted_data)
-        # else:
-        #     # Categorical selector.
-        #     _min, _max = self._get_min_max(targeted_data)
-
     @_log_call
     def build(self):
-        return self.selector.build()
+        # TODO: Align inner and outer ids to be handled in the same way as for other figure components.
+        selector_build_obj = self.selector.build()
+        return dcc.Loading(id=self.id, children=selector_build_obj) if self._dynamic else selector_build_obj
```

Comment:
Is the inner/outer id mechanism here aligned with how it's done for other Vizro figure components?

Reply:
I think yes, or at least it's the right way to do it. The best example to follow here would be `Figure`. You'll see that there's an extra `html.Div` in there, which from the comment I think is so it can be targeted by CSS but a temporary workaround before Dash fixes something.

So we can either do this the same way to match `Figure` (probably best if you add a comment about why) or this way, I don't mind.

Reply:
P.S. I think that in future with pattern-matching callbacks we might introduce a relationship between the filter `id` and the underlying selector `id` but that's for another time.

Reply:
I would leave `Filter.build` the same until Dash bugfix is released, which would enable the `html.Div(id=self.id, className="figure-container")` to be returned from `Filter.build` the same way it is done for our figure models.
I added a comment to enhance this implementation after the Dash bugfix.